internal/app/loki-multi-tenant-proxy/server.go

```diff
@@ -9,25 +9,49 @@ import (
 
 	"github.com/giantswarm/loki-multi-tenant-proxy/internal/pkg"
 	"github.com/urfave/cli/v2"
+	"go.uber.org/zap"
 )
 
 // Serve serves
 func Serve(c *cli.Context) error {
 	lokiServerURL, _ := url.Parse(c.String("loki-server"))
-	serveAt := fmt.Sprintf(":%d", c.Int("port"))
+	addr := fmt.Sprintf(":%d", c.Int("port"))
 	authConfigLocation := c.String("auth-config")
 	authConfig, _ := pkg.ParseConfig(&authConfigLocation)
 	authConfig.KeepOrgID = c.Bool("keep-orgid")
 
-	http.HandleFunc("/", createHandler(lokiServerURL, authConfig))
-	if err := http.ListenAndServe(serveAt, nil); err != nil {
-		log.Fatalf("Loki multi tenant proxy can not start %v", err)
+	logLevel := c.String("log-level")
+	if logLevel == "" {
+		logLevel = "INFO"
+	}
+
+	zapConfig := zap.NewProductionConfig()
+	level, err := zap.ParseAtomicLevel(logLevel)
+	if err != nil {
+		panic(fmt.Sprintf("Could not parse log level %v", zap.Error(err)))
+	}
+	zapConfig.Level = level
+
+	logger := zap.Must(zapConfig.Build())
+	defer logger.Sync()
+
+	errorLogger, err := zap.NewStdLogAt(logger, zap.ErrorLevel)
+	if err != nil {
+		logger.Fatal("Can not create error logger", zap.Error(err))
```

Comment:
I know `Fatal` would interrupt execution, but being explicit helps with consistency.

```suggestion
		logger.Print("Can not create error logger", zap.Error(err))
		return err
```

Reply:
Well I do not agree. We should know that logger.Fatal exists and using print makes it an info log, not a fatal one

Reply:
Fixed